airflow/providers/cncf/kubernetes/triggers/pod.py

```diff
@@ -155,18 +155,32 @@ def serialize(self) -> tuple[str, dict[str, Any]]:
             },
         )
 
+    def _get_terminal_event(self, state) -> TriggerEvent:
+        if state == PodPhase.SUCCEEDED:
+            status = "success"
+        else:
+            status = "failed"
+        return TriggerEvent({"status": status, "namespace": self.pod_namespace, "name": self.pod_name})
+
     async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
         """Get current pod status and yield a TriggerEvent."""
         self.log.info("Checking pod %r in namespace %r.", self.pod_name, self.pod_namespace)
         try:
             state = await self._wait_for_pod_start()
             if state in PodPhase.terminal_states:
-                event = TriggerEvent(
-                    {"status": "done", "namespace": self.pod_namespace, "pod_name": self.pod_name}
-                )
+                event = self._get_terminal_event(state)
             else:
                 event = await self._wait_for_container_completion()
             yield event
+        except PodLaunchTimeoutException as e:
+            description = self._format_exception_description(e)
+            yield TriggerEvent(
+                {
+                    "status": "timeout",
+                    "error_type": e.__class__.__name__,
+                    "description": description,
+                }
+            )
```

Comment:
In the _wait_for_pod_start method, we should change
`start_time = timezone.utcnow()` to `start_time = datetime.datetime.now(tz=datetime.timezone.utc)`

Reply:
Also should we use the output of `define_container_state` in `_wait_for_pod_start` and check for container statuses too?

Reply:
changed `timezone.utcnow()` to `datetime.datetime.now(tz=datetime.timezone.utc)`
now `_wait_for_pod_start ` return define_container_state